assets/js/apps/search/components/filters/filterSlice.jsx

```diff
@@ -0,0 +1,356 @@
+                    updateActiveSchemasReducer(state, {
+                        payload: {
+                            typeId: target[0],
+                            value: null
+                        }
+                    })
+                } else {
+                    updateTypeAttributeReducer(state, {
+                        payload: {
+                            typeId: target[0],
+                            attributeId: target[1],
+                            value: null
+                        }
+                    })
+                }
+                break;
+            case "schemaParameter":
+                updateSchemaParameterReducer(state, {
+                    payload: {
+                        schemaId: target[0],
+                        parameterId: target[1],
+                        value: null
+                    }
+                })
+                break;
+        }
+    });
+};
+
+const filters = createSlice({
+    name: 'filters',
+    initialState,
+    reducers: {
+        getFiltersStart: (state, { payload }) => {
+            state.isLoading = true;
+            state.error = null;
+        },
+        getFiltersSuccess: (state, { payload }) => {
+            state.schemas = mapSchemasById(payload);
+            state.typeSchemas = mapSchemaIdToType(payload);
+            state.isLoading = false;
+            state.error = null;
+        },
+        getFiltersFailure: (state, { payload:error }) => {
+            console.error("Error", error);
+            state.isLoading = false;
+            state.error = error.toString();
+        },
+        updateTypeAttribute: updateTypeAttributeReducer,
+        updateSchemaParameter: updateSchemaParameterReducer,
+        updateActiveSchemas: updateActiveSchemasReducer,
+        resetFilters: resetFiltersReducer,
+        updateFiltersByQuery: (state, { payload:filterQuery }) => {
+            // First, clear all previous filters to start off with a fresh slate.
+            // TODO Optimise this if necessary when there are a lot of filters.
+            resetFiltersReducer(state);
+            if (!filterQuery || !filterQuery.content) {
+                return;
+            }
+            // Take the filter list from the content of the root "and" query.
+            const filterList = filterQuery.content;
+            filterList.forEach(filter => {
+                const {kind ,target, content, op} = filter,
+                    value = {op, content};
+                let currValue, newValue;
+                switch (kind) {
+                    // TODO Refactor this function so that logic for 
+                    // adding filter values to existing can be reused
+                    // by both typeAttribute and schemaParameter.
+                    case "typeAttribute":
+                        // We may have added value for this field earlier in the iteration.
+                        currValue = typeAttrSelector(state,target[0],target[1]).value;
```

Comment:
Codacy has a fix for the issue: [A space is required after ','.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933948)
```suggestion
                        currValue = typeAttrSelector(state, target[0],target[1]).value;
```